### PR TITLE
mgr/dashboard: remove usage of 'rgw_frontend_ssl_key'

### PR DESCRIPTION
**Pull request by avanthakkar**
Fixes: https://tracker.ceph.com/issues/51643
Signed-off-by: Avan Thakkar <athakkar@redhat.com>

Removing the usage of rgw_frontend_ssl_key from the rgw service form.

Before:
Screenshot from 2021-07-13 18-16-13

After:
Screenshot from 2021-07-13 18-08-17

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html [hunk 1]
```diff
@@ -508,7 +508,7 @@
           </div>
 
           <!-- ssl_key -->
-          <div *ngIf="serviceForm.controls.ssl.value"
+          <div *ngIf="serviceForm.controls.ssl.value && ['iscsi', 'ingress'].includes(serviceForm.controls.service_type.value)"
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html:511]: Just curious, why not something like `serviceForm.controls.service_type.value !== 'rgw'` or something like that? I mean check whether the type is not rgw and then show the field?

**Review by nizamial09 — APPROVED** [review]
LGTM! apart from one small question. Thanks @avanthakkar

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html (hunk 1 above)
Comment (avanthakkar) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html:511]: Oh yes..I was looking for the same..thanks @nizamial09 . Yeah it makes more sense to exclude here!

**Comment by avanthakkar** [comment]
> LGTM! apart from one small question. Thanks @avanthakkar (probably a unit test to make sure its not showing for type rgws)

But we don't pass the ssl_key itself so Idk what to test here. It's just the visual difference you will see as I have attached the screenshot in PR description.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html (hunk 1 above)
Comment (avanthakkar) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html]: Done.

**Comment by nizamial09** [comment]
> what to test here

sorry i wasnt being clear, I was actually asking to test whether the field is present or not. Something like this in the rgw test area of the service component.
```

      it('should not show private key field', () => {
        formHelper.setValue('ssl', true);
        fixture.detectChanges();
        const ssl_key = fixture.debugElement.query(By.css('#ssl_key'));
        expect(ssl_key).toBeNull();
      })
 ```

**Comment by avanthakkar** [comment]
> > what to test here
> 
> @avanthakkar sorry i wasnt being clear, I was actually asking to test whether the field is present or not. Something like this in the rgw test area of the service component.
> 
> ```
> 
>       it('should not show private key field', () => {
>         formHelper.setValue('ssl', true);
>         fixture.detectChanges();
>         const ssl_key = fixture.debugElement.query(By.css('#ssl_key'));
>         expect(ssl_key).toBeNull();
>       })
> ```

Yes you're right. Thanks @nizamial09 , updated the changes.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html [hunk 2]
```diff
@@ -508,7 +508,7 @@
           </div>
 
           <!-- ssl_key -->
-          <div *ngIf="serviceForm.controls.ssl.value"
+          <div *ngIf="serviceForm.controls.ssl.value && serviceForm.controls.service_type.value !== 'rgw'"
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html:511]: Shouldn't we somehow indicate the user that they're expected to submit a concatenated certificate + key (.PEM-like format)?

**Review by epuertat — APPROVED** [review]
LGTM Thanks @avanthakkar !

**Pull request merged**